src/Doctrine/DoctrineHelper.php

```diff
@@ -161,16 +161,32 @@ public function getMetadata(string $classOrNamespace = null, bool $disconnected
                 }
             }
 
-            foreach ($cmf->getAllMetadata() as $m) {
-                if (null === $classOrNamespace) {
-                    $metadata[$m->getName()] = $m;
-                } else {
-                    if ($m->getName() === $classOrNamespace) {
-                        return $m;
-                    }
+            try {
+                foreach ($cmf->getAllMetadata() as $m) {
+                    if (null === $classOrNamespace) {
+                        $metadata[$m->getName()] = $m;
+                    } else {
+                        if ($m->getName() === $classOrNamespace) {
+                            return $m;
+                        }
 
-                    if (0 === strpos($m->getName(), $classOrNamespace)) {
+                        if (0 === strpos($m->getName(), $classOrNamespace)) {
+                            $metadata[$m->getName()] = $m;
+                        }
+                    }
+                }
+            } catch (\ReflectionException $exception) {
+                foreach ($cmf->getLoadedMetadata() as $m) {
+                    if (null === $classOrNamespace) {
                         $metadata[$m->getName()] = $m;
+                    } else {
+                        if ($m->getName() === $classOrNamespace) {
+                            return $m;
+                        }
+
+                        if (0 === strpos($m->getName(), $classOrNamespace)) {
+                            $metadata[$m->getName()] = $m;
+                        }
```

Comment:
So, we wrap everything in a catch for `ReflectionException`. Can you explain what's happening here? What does that exception say?